src/main/java/edu/ucla/library/libcal/handlers/ProxyHandler.java

```diff
@@ -0,0 +1,121 @@
+
+package edu.ucla.library.libcal.handlers;
+
+import static edu.ucla.library.libcal.MediaType.APPLICATION_JSON;
+import static edu.ucla.library.libcal.MediaType.TEXT_PLAIN;
+import static info.freelibrary.util.Constants.EMPTY;
+import static info.freelibrary.util.Constants.SLASH;
+
+import info.freelibrary.util.HTTP;
+
+import edu.ucla.library.libcal.Constants;
+import edu.ucla.library.libcal.JsonKeys;
+import edu.ucla.library.libcal.MessageCodes;
+import edu.ucla.library.libcal.services.LibCalProxyService;
+import edu.ucla.library.libcal.services.OAuthTokenService;
+
+import info.freelibrary.util.Logger;
+import info.freelibrary.util.LoggerFactory;
+
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * A handler that processes status information requests.
+ */
+@SuppressWarnings("PMD.AvoidCatchingGenericException")
+public class ProxyHandler implements Handler<RoutingContext> {
+
+    /**
+     * The handler's logger.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProxyHandler.class, MessageCodes.BUNDLE);
+
+    /**
+     * A constant for the "?" to lead an HTTP query string.
+     */
+    private static final String QUESTION_MARK = "?";
+
+    /**
+     * The handler's copy of the Vert.x instance.
+     */
+    private final Vertx myVertx;
+
+    /**
+     * A service for LibCal API calls
+     */
+    private final LibCalProxyService myApiProxy;
+
+    /**
+     * A service for LibCal OAuth calls
+     */
+    private final OAuthTokenService myTokenProxy;
+
+    /**
+     * Creates a handler that returns a status response.
+     *
+     * @param aVertx A Vert.x instance
+     */
+    public ProxyHandler(final Vertx aVertx) {
+        myVertx = aVertx;
+        myApiProxy = LibCalProxyService.createProxy(myVertx);
+        myTokenProxy = OAuthTokenService.createProxy(myVertx);
+    }
+
+    @Override
+    public void handle(final RoutingContext aContext) {
+        final HttpServerResponse response = aContext.response();
+        final String path = aContext.request().path();
+
+        try {
+            if (path.equals(EMPTY) || path.equals(SLASH)) {
```

Comment:
It isn't possible for the path to equal the empty string; even if you call a WebClient on the "path" `""`, it converts that to `"/"`.

Even if we just checked for the root path, though, this if-block is getting us into request validation territory, which I don't think our application should be concerned with.

Reply:
What's so horrible about request validation? Shouldn't we tell the client it's sending a bad request?

Reply:
Request validation is good and well, but for this app would require more than simply checking that the path isn't `/`. I mean, there is a very large number of paths besides `/` that don't represent a resource in the LibCal API. Why not also make sure the path doesn't start with `/admin`? `/asdf`? All paths (i.e., resource identifiers) that aren't in the set defined at https://calendar.library.ucla.edu/admin/api/1.1 are equally invalid; `/` is not special in this way. Requests to these paths (likely) all return HTML; that fact alone would make it obvious (IMO) to an API client, expecting a JSON response representing the state of a resource, that it was making a bad request.

Since writing an OpenAPI spec for all LibCal endpoints (57 of them by my count) would be tedious, we decided that request validation should be the concern of LibCal.

Reply:
Removed validation logic